File: WebContent/cqut/hrms/view/organization/js/OrganizationService.js
var organizationApp = angular.module("organizationApp", []);

organizationApp.factory('OrganizationService',['$resource',function($resource){
	
	var organizationService = $resource('/Organization/:action', {});
	
	organizationService.insertOrganization=function(organization,sucesscb,errorcb){
		
		organizationService.save({action:"insertOrganization"},organization,sucesscb,errorcb);
	}
	
	organizationService.getOrganizations=function(sucesscb,errorcb){
		
		organizationService.query({action:"getOrganizations"},sucesscb,errorcb);
	}
	
	organizationService.getOrganizationById=function(EId,sucesscb,errorcb){
		
		organizationService.save({action:"getOrganizationById"},EId,sucesscb,errorcb);
	}
	
	organizationService.getOrganizationByName=function(name,sucesscb,errorcb){
		
		organizationService.save({action:"getOrganizationByName"},name,sucesscb,errorcb);
	}
	
	organizationService.deleteOrganization=function(EId,sucesscb,errorcb){
		
		organizationService.save({action:"deleteOrganization"},EId,sucesscb,errorcb);
	}
	
	organizationService.updateOrganization=function(organization,sucesscb,errorcb){
		
		organizationService.save({action:"updateOrganization"},organization,sucesscb,errorcb);
	}
	
	return organizationService;
	
}]);

File: src/com/cqut/service/organization/OrganizationService.java
package com.cqut.service.organization;

import java.util.List;

import com.cqut.entity.Organization;

public interface OrganizationService {

	public int insertOrganization(Organization organization);
	
	public List<Organization> getOrganizations(String where);
	
	public Organization getOrganizationById(int id);
	
	public List<Organization> getOrganizationByName(String name);
	
	public int deleteOrganization(int EId);
	
	public int updateOrganization(Organization organization);
}
